plugin/neomake.vim

```diff
@@ -1,37 +1,11 @@
 " vim: ts=4 sw=4 et
 
-function! s:NeomakeCommand(file_mode, enabled_makers)
-    if a:file_mode
-        call neomake#Make({
-            \ 'enabled_makers': len(a:enabled_makers) ?
-                \ a:enabled_makers :
-                \ neomake#GetEnabledMakers(&ft),
-            \ 'ft': &ft,
-            \ 'file_mode': 1,
-            \ })
-    else
-        call neomake#Make({
-            \ 'enabled_makers': len(a:enabled_makers) ?
-                \ a:enabled_makers :
-                \ neomake#GetEnabledMakers()
-            \ })
-    endif
-endfunction
-
-function! s:NeomakeSh(sh_command)
-    let custom_maker = neomake#utils#MakerFromCommand(&shell, a:sh_command)
-    let custom_maker.name = 'sh: '.a:sh_command
-    let custom_maker.remove_invalid_entries = 0
-    let enabled_makers =  [custom_maker]
-    call neomake#Make({'enabled_makers': enabled_makers})
-endfunction
-
-command! -nargs=* -bang -bar -complete=customlist,neomake#CompleteMakers Neomake call s:NeomakeCommand(<bang>1, [<f-args>])
+command! -nargs=* -bang -bar -complete=customlist,neomake#CompleteMakers Neomake call neomake#WithCallback(<bang>1, [<f-args>])
 " These commands are available for clarity
 command! -nargs=* -bar -complete=customlist,neomake#CompleteMakers NeomakeProject Neomake! <args>
 command! -nargs=* -bar -complete=customlist,neomake#CompleteMakers NeomakeFile Neomake <args>
 
-command! -nargs=+ -complete=shellcmd NeomakeSh call s:NeomakeSh(<q-args>)
+command! -nargs=+ -complete=shellcmd NeomakeSh call neomake#ShWithCallback(<q-args>)
```

Comment:
Callbacks are never meant to be passed through the commands. Since `<q-args>` creates only a single argument, and `<f-args>` are surrounded by square brackets, the third argument will always be empty when invoked from the commands `:Neomake` and `:NeomakeSh`.

Callbacks can only be requested by directly calling `neomake#WithCallback` and `neomake#ShWithCallback`.